ROVU/Core/src/utility/SimulatorMonitor.java

```diff
@@ -1,31 +1,83 @@
 package utility;
 
+import java.awt.geom.Rectangle2D;
 import java.util.ArrayList;
+
 import java.util.List;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
+import model.Environment;
+import model.Hospital;
 import model.Mission;
 import model.MissionPoint;
 import project.AbstractRobotSimulator;
 import project.AbstractSimulatorMonitor;
 import project.Point;
+import robot.A_Star;
+import model.Node;
 import robot.RobotHandler;
+import simbad.sim.Boundary;
 import simbad.sim.EnvironmentDescription;
+import simbad.sim.Wall;
 
 public class SimulatorMonitor extends AbstractSimulatorMonitor<RobotHandler> {
+	Point [] com ;
+	public  List <Boundary> bounds;
+	public  List <Wall> walls;
+
+	
 	Set<RobotHandler> r;
-	public SimulatorMonitor(Set<RobotHandler> robots, EnvironmentDescription e) {
+	public SimulatorMonitor(Set<RobotHandler> robots, EnvironmentDescription e,List <Boundary> bounds, List <Wall> walls) {
 		super(robots, e);
 		r = robots;
+		this.walls=walls;
+		this.bounds=bounds;
 	}
 
-	
+
 
 	public void update(RobotHandler arg0) {
-//		System.out.println(arg0.getName());
-//		System.out.println(arg0.getPosition());
-//
+		
+		/*
```

Comment:
I know this part is currently commented out, but it's super hard to follow the code if the indentation is not structured. For example, I don't know where the `if`-statement ends, not where the `for`-loop starts and ends.